### DISCUSSION TITLE
How to immediately end training?

### DISCUSSION THREAD

**Discussion by vedantroy** [Lightning-AI/pytorch-lightning#13563] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I know you can end training by returning `-1` in `on_train_batch_start`, or whatever the callback is called, but this can take a while to finish if you have a lot of epochs.

Is there a way to tell Pytorch Lightning to immediately end training?

**Comment by treblenalto** [comment]
If your purpose is to debug training process,  try `Trainer(fast_dev_run=True)` or you can shorten epochs by using fraction of data with `Trainer(limit_train_batches=0.1, limit_val_batches=0.01)`
